pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2466,32 +2462,73 @@ private String receiveCommandStatus() throws IOException {
     return status;
   }
 
+  private static boolean isDigitAt(String status, int i) {
+    return i > 0 && i < status.length() && Character.isDigit(status.charAt(i));
+  }
+
+  private static int digitAt(String s, int pos) {
+    int c = s.charAt(pos) - '0';
+    if (c < 0 || c > 9) {
+      throw new NumberFormatException("Input string: \"" + s + "\"");
+    }
+    return c;
+  }
+
+  /**
+   * Faster version of {@link Long#parseLong(String)} when parsing a substring is required
+   *
+   * @param s string to parse
+   * @param beginIndex begin index
+   * @param endIndex end index
+   * @return long value
+   */
+  private static long toLong(String s, int beginIndex, int endIndex) {
+    // Fallback to default implementation in case the string is long or the first digi
```

Comment:
I’m guessing the long string is to better handle overflow. Why do we want default implementation for first char not a digit? Don’t both result in number format exception?

Reply:
`isDigit` was not needed there. The idea is just delegate large strings to default implementation to avoid dealing with things like MAX_LONG, etc.

Reply:
Just one more thing: I assume all the numbers are positive, so there cannot be something like `-42`